homeassistant/components/tado/switch.py

```diff
@@ -0,0 +1,99 @@
+"""Module for Tado child lock switch entity."""
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TadoConfigEntry
+from .const import TADO_CHILD_LOCK_TRANSLATION_KEY
+from .entity import TadoDataUpdateCoordinator, TadoZoneEntity
+
+
+class TadoChildLockSwitchEntity(TadoZoneEntity, SwitchEntity):
+    """Representation of a Tado child lock switch entity."""
+
+    _attr_unique_id: str | None = None
+
+    def __init__(
+        self,
+        coordinator: TadoDataUpdateCoordinator,
+        zone_name: str,
+        zone_id: int,
+        device_info: dict[str, Any],
+    ) -> None:
+        """Initialize the Tado child lock switch entity."""
+        super().__init__(zone_name, coordinator.home_id, zone_id, coordinator)
+
+        self._device_info = device_info
+        self._device_id = self._device_info["shortSerialNo"]
+        self._state: bool | None = None
+        self._attr_unique_id = f"{zone_name}-child-lock"
```

Comment:
unique id needs to be stable. I assume that `zone_name` can be changed by the user and is not stable, right?
the `climate` platform uses `self._attr_unique_id = f"{zone_type} {zone_id} {coordinator.home_id}"` so I suggest using a similar id here

Reply:
No it wasn't stable and I didn't understand what it did, until I changed the zone name and realized. Thanks for the warning.

I set it according to zone id and home id now, as I don't think the zone type is necessary here. If the underyling device would change, this would still use the same switch entity. Only when the zone id changes (tado wise) this would lead to "duplicate" entities. But this applies to other entities in home assistant tado platform as well